libs/SmartStore/src/com/salesforce/androidsdk/smartstore/app/SmartStoreUpgradeManager.java

```diff
@@ -82,4 +89,21 @@ protected synchronized void upgradeSmartStore() {
     public String getInstalledSmartStoreVersion() {
         return getInstalledVersion(SMART_STORE_KEY);
     }
+
+    @Override
+    protected void upgradeTo8Dot2() {
+        super.upgradeTo8Dot2();
+        final Map<String, DBOpenHelper> dbOpenHelperMap = DBOpenHelper.getOpenHelpers();
+        if (dbOpenHelperMap != null) {
+            final Collection<DBOpenHelper> dbOpenHelpers = dbOpenHelperMap.values();
+            for (final DBOpenHelper dbOpenHelper : dbOpenHelpers) {
+                if (dbOpenHelper != null) {
+                    final String oldEncryptionKey = SmartStoreSDKManager.getLegacyEncryptionKey();
```

Comment:
Migration steps for all `SmartStore` data. Uses `PRAGMA REKEY` behind the scenes, available in `SQLCipher`.